test/integration/AuthenticatedFileResourceStore.test.ts

```diff
@@ -0,0 +1,238 @@
+import { promises as fs } from 'fs';
+import * as url from 'url';
+import { MockResponse } from 'node-mocks-http';
+import { HttpHandler, ResourceStore } from '../..';
+import { AuthenticatedFileResourceStoreConfig } from '../configs/AuthenticatedFileResourceStoreConfig';
+import { call, callFile, setAcl } from '../util/Util';
+
+describe('A server using a FileResourceStore', (): void => {
+  let config: AuthenticatedFileResourceStoreConfig;
+  let handler: HttpHandler;
+  let store: ResourceStore;
+
+  beforeAll(
+    async(): Promise<void> => {
+      config = new AuthenticatedFileResourceStoreConfig();
+      handler = config.getHandler();
+      ({ store } = config);
+
+      await setAcl(
+        store,
+        'http://test.com/',
+        { read: true, write: true, append: true },
+        true,
+        true,
+        true,
+        undefined,
+        'agent',
+      );
+
+      // Check if permanent file exists, if not make one
+      let requestUrl = new URL('http://test.com/permanent.txt');
+      let response = await call(
+        handler,
+        requestUrl,
+        'GET',
+        { accept: 'text/*' },
+        [],
+      );
+      if (response.statusCode !== 200) {
+        requestUrl = new URL('http://test.com/');
+        const fileData = await fs.readFile('test/testfiles/permanent.txt');
+
+        response = await callFile(
+          handler,
+          requestUrl,
+          'POST',
+          {
+            'content-type': 'application/octet-stream',
+            slug: 'permanent.txt',
+            'transfer-encoding': 'chunked',
+          },
+          fileData,
+        );
+        expect(response.statusCode).toBe(200);
+      }
+    },
+  );
+  afterAll(
```

Comment:
The problem is that other tests could be still running. So if the directory gets cleaned completely before all others are done errors will probably occur.

Reply:
Which other tests? `afterAll` is supposed to execute when all tests are finished right? Unless you mean tests in other files, but maybe you can then have a separate folder for each of those tests so this is not an issue? Or somehow have a wrapper around all those tests that takes care of this (this might be more work though).

Reply:
This `afterAll` is scoped for the `AuthenticatedFileResourceStore` tests. But by changing the configs its possible to have a wrapperfolder that can get deleted after all tests ran.

Reply:
It's just that this could leave files behind when certain tests fail (or even when there's something wrong with the store and the delete doesn't work) which could be annoying for the user and might even cause the next tests to run incorrectly (since there are already files there before starting).

Reply:
I'm trying to get a `globalTeardown` with jest but it does not work for the moment.